src/effects/Tracker.cpp

```diff
@@ -72,96 +91,159 @@ void Tracker::init_effect_details()
-        }
+        if (trackedData->draw_box.GetValue(frame_number) == 1) 
+		{
+			std::vector<int> stroke_rgba = trackedData->stroke.GetColorRGBA(frame_number);
+			int stroke_width = trackedData->stroke_width.GetValue(frame_number);
+			float stroke_alpha = trackedData->stroke_alpha.GetValue(frame_number);
+			std::vector<int> bg_rgba = trackedData->background.GetColorRGBA(frame_number);
+			float bg_alpha = trackedData->background_alpha.GetValue(frame_number);
+
+			// Create a rotated rectangle object that holds the bounding box
+			cv::RotatedRect box ( cv::Point2f( (int)(fd.cx*fw), (int)(fd.cy*fh) ), 
+								  cv::Size2f( (int)(fd.width*fw), (int)(fd.height*fh) ), 
+								  (int) (fd.angle) );
+
+			DrawRectangleRGBA(frame_image, box, bg_rgba, bg_alpha, 1, true);
+			DrawRectangleRGBA(frame_image, box, stroke_rgba, stroke_alpha, stroke_width, false);
+		}
+		
+		// Get the image of the Tracked Object' child clip
+		if (trackedData->ChildClipId() != ""){
+			// Cast the parent timeline of this effect 
+			Timeline* parentTimeline = (Timeline *) ParentTimeline();
+			if (parentTimeline){
+				// Get the Tracked Object's child clip
+				Clip* childClip = parentTimeline->GetClip(trackedData->ChildClipId());
+				if (childClip){
+					// Get the image of the child clip for this frame
+					std::shared_ptr<Frame> f(new Frame(1, frame->GetWidth(), frame->GetHeight(), "#00000000"));
+					std::shared_ptr<Frame> childClipFrame = childClip->GetFrame(f, frame_number);
+					childClipImage = childClipFrame->GetImage();
+
+					// Set the Qt rectangle with the bounding-box properties
+					boxRect.setRect((int)((fd.cx-fd.width/2)*fw),
+									(int)((fd.cy - fd.height/2)*fh),
+									(int)(fd.width*fw),
+									(int)(fd.height*fh) );
+				}
+			}
+		}
+	
     }
 
 	// Set image with drawn box to frame
     // If the input image is NULL or doesn't have tracking data, it's returned as it came
 	frame->SetImageCV(frame_image);
 
-	return frame;
-}
-
-// Load protobuf data file
-bool Tracker::LoadTrackedData(std::string inputFilePath){
-	using std::ios;
+	// Set the bounding-box image with the Tracked Object's child clip image
+	if (childClipImage){
+		// Get the frame image
+		QImage frameImage = *(frame->GetImage());
 
-    // Create tracker message
-    pb_tracker::Tracker trackerMessage;
-
-    {
-        // Read the existing tracker message.
-        std::fstream input(inputFilePath, ios::in | ios::binary);
-        if (!trackerMessage.ParseFromIstream(&input)) {
-            std::cerr << "Failed to parse protobuf message." << std::endl;
-            return false;
-        }
-    }
+		// Set a Qt painter to the frame image
+		QPainter painter(&frameImage);
 
-    // Make sure the trackedData is empty
-    trackedDataById.clear();
+		// Draw the child clip image inside the bounding-box
+		painter.drawImage(boxRect, *childClipImage, QRectF(0, 0, frameImage.size().width(),  frameImage.size().height()));
```

Comment:
```suggestion
		painter.drawImage(boxRect, *childClipImage, QRectF(frameImage.rect()));
```

(Though, if you're drawing the entire image, I don't _think_ this is necessary. If I'm not mistaken, QImage's `drawImage()` defaults to drawing the entire source image in the area specified, scaled accordingly. So  that would make this equivalent):
```c++
painter.drawImage(boxRect, *childClipImage);
```